### ISSUE TITLE
ravendb_8761_can_group_by_array_content fails with 4.1.2

### ISSUE THREAD

**Issue by kjk** [ravendb/ravendb-go-client#68]
Used to work with 4.0.6.

https://ci.appveyor.com/project/ravendb/ravendb-go-client/builds/20134873
https://travis-ci.org/ravendb/ravendb-go-client/builds/452178281

**Comment by kjk** [comment]
Java (my fork of older java version) also fails with 4.1.2 on Windows:
```
 INFO: Executing query from Orders                                                                                     
 group by array(lines[].product), array(lines[].quantity)                                                             
 order by lines[].quantity                                                                                            
 select lines[].product as products, lines[].quantity as quantities, count() as count on index null in http://127.0.0.
1:7378 (DB: test_db_1);test_db_1                                                                                      
Dec 15, 2018 4:34:52 PM net.ravendb.client.documents.session.operations.QueryOperation ensureIsAcceptableAndSaveResult
                                                                                                                      
INFO: Query from Orders                                                                                               
 group by array(lines[].product), array(lines[].quantity)                                                             
 order by lines[].quantity                                                                                            
 select lines[].product as products, lines[].quantity as quantities, count() as count returned 1 results (total index 
results: 2)                                                                                                           
Dec 15, 2018 4:34:52 PM net.ravendb.client.documents.session.operations.QueryOperation logQuery                       
INFO: Executing query from Orders group by Array(lines[].product), Array(lines[].quantity) order by count select lines
[].product as products, lines[].quantity as quantities, count() as count on index null in http://127.0.0.1:7378 (DB: t
est_db_1);test_db_1                                                                                                   
Dec 15, 2018 4:34:52 PM net.ravendb.client.documents.session.operations.QueryOperation ensureIsAcceptableAndSaveResult
                                                                                                                      
INFO: Query from Orders group by Array(lines[].product), Array(lines[].quantity) order by cPROXY GET /cluster/topology
 HTTP/1.1                                                                                                             
ount select lines[].product as products, lines[].quantity as quantities, count() as count returned 2 results (total in
dex results: 2)                                                                                                       
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.54 sec <<< FAILURE! - in net.ravendb.client.test.ser
ver.documents.indexing.auto.RavenDB_8761                                                                              
can_group_by_array_content()  Time elapsed: 5.403 sec  <<< FAILURE!                                                   
java.lang.AssertionError:                                                                                             
                                                                                                                      
Expected size:<2> but was:<1> in:                                                                                     
<[net.ravendb.client.test.server.documents.indexing.auto.RavenDB_8761$ProductCount@6c4f9535]>                         
        at net.ravendb.client.test.server.documents.indexing.auto.RavenDB_8761.can_group_by_array_content(RavenDB_8761
.java:259)                                                                                                            
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                             
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)           
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)   
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)                                                 
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:389)                     
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:115)                    
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescri
ptor.java:167)                                                                                                        
        at org.junit.jupiter.engine.execution.ThrowableCollector.execute(ThrowableCollector.java:40)                  
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java
:163)                                                                                                                 
```

Assert:
```
                for (List<ProductCount> products : Arrays.asList(productCounts1, productCounts2)) {
                    assertThat(products)
                            .hasSize(2);
```

Query log for Java: https://gist.github.com/kjk/f0e7c98fc14e442c44bac196a6641987

Looks like this query returned 1 result but was expected to return 2:
```
=========== 11:
POST /databases/test_db_1/queries?queryHash=9c132814791bc4a4134b9adfeb498224 HTTP/1.1
Host: 127.0.0.1:7378
Transfer-Encoding: chunked
Client-Configuration-Etag: "1100310647005720893"
Content-Type: application/json; charset=UTF-8
Raven-Client-Version: 4.0.0
Topology-Etag: "-1"
User-Agent: Apache-HttpClient/4.5.1 (Java/10.0.2)

{
  "Query": "from Orders\n group by array(lines[].product), array(lines[].quantity)\n order by lines[].quantity\n select lines[].product as products, lines[].quantity as quantities, count() as count",
  "QueryParameters": {},
  "WaitForNonStaleResults": true,
  "WaitForNonStaleResultsTimeout": "00:00:15"
}
--------
HTTP/1.1 200 OK
Transfer-Encoding: chunked
Content-Type: application/json; charset=utf-8
Date: Sun, 16 Dec 2018 00:34:51 GMT
Etag: -739892925503178011
Raven-Server-Version: 4.1.2.41004
Request-Time: 54
Server: Kestrel
Vary: Accept-Encoding

{
  "DurationInMs": 54,
  "IncludedPaths": null,
  "Includes": {},
  "IndexName": "Auto/Orders/ByCountReducedByArray(lines[].product)AndArray(lines[].quantity)",
  "IndexTimestamp": "2018-12-16T00:34:52.2174935",
  "IsStale": false,
  "LastQueryTime": "2018-12-16T00:34:52.2135587",
  "NodeTag": "A",
  "ResultEtag": -739892925503178000,
  "Results": [
    {
      "@metadata": {
        "@change-vector": null,
        "@index-score": 1,
        "@projection": true
      },
      "count": 2,
      "products": [
        "products/1",
        "products/2"
      ],
      "quantities": [
        1,
        2
      ]
    }
  ],
  "SkippedResults": 0,
  "TotalResults": 2
}
```

Was it something that was fixed after 4.1.2 server?
Was it something that was fixed in Java client?

**Comment by kjk** [comment]
seems to be working with 4.1.3 server
